geopandas/tools/tests/test_sjoin.py

```diff
@@ -82,6 +82,23 @@ def test_sjoin_values(self):
         df = sjoin(self.polydf, self.pointdf, how='left')
         self.assertEquals(df.shape, (12,8))
 
+    def test_no_overlapping_geometry(self):
+        # Note: these tests are for correctly returning GeoDataFrame
+        # when result of the join is empty
+
+        subset_pointdf = self.pointdf.iloc[17:]
+
+        df_inner = sjoin(subset_pointdf, self.polydf, how='inner')
+        self.assertEquals(df_inner.shape, (0,8))
```

Comment:
I have updated with a new commit with a little extra testing for the 'inner' join.